src/main/io/osd.c

```diff
@@ -180,305 +178,285 @@ static void osdDrawSingleElement(uint8_t item)
-            for (int8_t y = -hudheight; y <= hudheight; y++) {
-                displayWriteChar(osdDisplayPort, elemPosX - hudwidth, elemPosY + y, SYM_AH_DECORATION);
-                displayWriteChar(osdDisplayPort, elemPosX + hudwidth, elemPosY + y, SYM_AH_DECORATION);
-            }
+        osdDrawSingleElement(OSD_HORIZON_SIDEBARS);
 
-            // AH level indicators
-            displayWriteChar(osdDisplayPort, elemPosX - hudwidth + 1, elemPosY, SYM_AH_LEFT);
-            displayWriteChar(osdDisplayPort, elemPosX + hudwidth - 1, elemPosY, SYM_AH_RIGHT);
+        return;
+    }
 
-            return;
-        }
+    case OSD_HORIZON_SIDEBARS:
+    {
+        elemPosX = 14;
+        elemPosY = 6;
 
-        case OSD_ROLL_PIDS:
-        {
-            const pidProfile_t *pidProfile = currentPidProfile;
-            tfp_sprintf(buff, "ROL %3d %3d %3d", pidProfile->pid[PID_ROLL].P, pidProfile->pid[PID_ROLL].I, pidProfile->pid[PID_ROLL].D);
-            break;
+        if (displayScreenSize(osdDisplayPort) == VIDEO_BUFFER_CHARS_PAL) {
+            ++elemPosY;
         }
 
-        case OSD_PITCH_PIDS:
-        {
-            const pidProfile_t *pidProfile = currentPidProfile;
-            tfp_sprintf(buff, "PIT %3d %3d %3d", pidProfile->pid[PID_PITCH].P, pidProfile->pid[PID_PITCH].I, pidProfile->pid[PID_PITCH].D);
-            break;
+        // Draw AH sides
+        int8_t hudwidth = AH_SIDEBAR_WIDTH_POS;
+        int8_t hudheight = AH_SIDEBAR_HEIGHT_POS;
+        for (int8_t y = -hudheight; y <= hudheight; y++) {
+            displayWriteChar(osdDisplayPort, elemPosX - hudwidth, elemPosY + y, SYM_AH_DECORATION);
+            displayWriteChar(osdDisplayPort, elemPosX + hudwidth, elemPosY + y, SYM_AH_DECORATION);
         }
 
-        case OSD_YAW_PIDS:
-        {
-            const pidProfile_t *pidProfile = currentPidProfile;
-            tfp_sprintf(buff, "YAW %3d %3d %3d", pidProfile->pid[PID_YAW].P, pidProfile->pid[PID_YAW].I, pidProfile->pid[PID_YAW].D);
-            break;
-        }
+        // AH level indicators
+        displayWriteChar(osdDisplayPort, elemPosX - hudwidth + 1, elemPosY, SYM_AH_LEFT);
+        displayWriteChar(osdDisplayPort, elemPosX + hudwidth - 1, elemPosY, SYM_AH_RIGHT);
 
-        case OSD_POWER:
-        {
-            tfp_sprintf(buff, "%dW", getAmperage() * getBatteryVoltage() / 1000);
-            break;
-        }
+        return;
+    }
 
-        case OSD_PIDRATE_PROFILE:
-        {
-            const uint8_t pidProfileIndex = getCurrentPidProfileIndex();
-            const uint8_t rateProfileIndex = getCurrentControlRateProfileIndex();
-            tfp_sprintf(buff, "%d-%d", pidProfileIndex + 1, rateProfileIndex + 1);
-            break;
-        }
+    case OSD_ROLL_PIDS:
+    {
+        const pidProfile_t *pidProfile = currentPidProfile;
+        tfp_sprintf(buff, "ROL %3d %3d %3d", pidProfile->pid[PID_ROLL].P, pidProfile->pid[PID_ROLL].I, pidProfile->pid[PID_ROLL].D);
```

Comment:
this may be collected into function (`sprintPID(char* buff, const char* name, pid8_t * pid);`)